api/admin/controller/__init__.py

```diff
@@ -386,7 +388,11 @@ def __call__(self, collection, book, path=None):
         if not setting_up:
             admin = self.authenticated_admin_from_request()
             if isinstance(admin, ProblemDetail):
-                redirect_url = flask.request.url
+                url = urlparse(flask.request.url)
```

Comment:
At some point, flask started returning the unescaped URL in `flask.request.url`, however we were expected the escaped URL here. So the new code does this escaping.

Reply:
Don't url.query and fragment need to be quoted too?

Reply:
Good catch, yes. Those are quoted now.